core/bindings.rs

```diff
@@ -352,8 +353,10 @@ fn print(
   };
   if is_err {
     eprint!("{}", str_.to_rust_string_lossy(tc_scope));
+    stdout().flush().unwrap();
   } else {
     print!("{}", str_.to_rust_string_lossy(tc_scope));
+    stdout().flush().unwrap();
```

Comment:
I added flush here because print! macro doesn't seem flushing automatically. https://github.com/rust-lang/rust/issues/23818

Reply:
@kt3k this is a bit strange - `Deno.core.print()` works for me without a problem without the `flush()` and it's one of purposes of `Deno.core.print` not to include new line by default. Could you explain the reason for this change?

Reply:
@bartlomieju `print!` is line-buffered as discussed in https://github.com/rust-lang/rust/issues/23818. So if we use it like `Deno.core.print('hello')`, the string doesn't show in stdout until new line is printed or process is finished. (Try the script `Deno.core.print('xyz'); setTimeout(() => {}, 4000);` for example).

If we don't add `flush()` here, `prompt` works like the below:

```
> prompt('hi!')
hello
hi! "hello"
```

(The answer and question are reversed here.) I think this isn't what the users expect.

If we add `flush()`, it works like:

```
> prompt('hi!')
hi! hello
"hello"
```

Reply:
@kt3k I can not reproduce that:
![image](https://user-images.githubusercontent.com/7829205/97783994-acec9a00-1b9b-11eb-8848-f0f84d9c3931.png)

Reply:
In repl, the returned value is printed with line break (such as "1\n") and this line break should trigger flushing of stdout.

Try the following:

```
deno eval "Deno.core.print('xyz'); setTimeout(() => {}, 4000);"
```